src/logic/credentials-storage.ts

```diff
@@ -0,0 +1,22 @@
+import { UserCredentials } from '../types/types';
+import { TokenBearer } from '../types/types';
+
+export function storeToken(token: TokenBearer) {
+  localStorage.setItem('tokenBearer', JSON.stringify(token));
+}
+
+export function retrieveToken(): TokenBearer {
+  return JSON.parse(
+    localStorage.getItem('tokenBearer') as string,
+  ) as TokenBearer;
+}
+
+export function storeUserCredentials(credentials: UserCredentials) {
+  localStorage.setItem('userCredentials', JSON.stringify(credentials));
+}
+
+export function retrieveUserCredentials(): UserCredentials {
+  return JSON.parse(
+    localStorage.getItem('userCredentials') as string,
+  ) as UserCredentials;
+}
```

Comment:
Likewise, the retrieve functions could look lile this:
```suggestion
export function retrieveUserCredentials(rootUrl: string): UserCredentialsResponse | null {
  const allCredentialsAsString = localStorage.userCredentials;
  if (!allCredentialsAsString) {
    return null;
  }
  
  const allCredentials =  JSON.parse(allCredentialsAsString) as UserCredentials;
  return allCredentials[rootUrl] ?? null;
}
```

and similarly for the other retrieve function.